lib/remotable/nosync.rb

```diff
@@ -1,37 +1,84 @@
+require "active_resource/threadsafe_attributes"
+
 module Remotable
   module Nosync
 
-
-    def nosync!
-      self.nosync = true
+    def self.included(base)
+      base.include InstanceMethods
     end
 
-    def nosync(new_value=true)
-      old_value = _nosync
-      self.nosync = new_value
-      yield
-    ensure
-      @nosync = old_value
+    def self.extended(base)
+      base.extend ClassMethods
     end
 
-    def nosync=(val)
-      @nosync = val
-    end
+    module InstanceMethods
 
-    def nosync_value?
-      !_nosync.nil?
-    end
+      def nosync!
+        self.nosync = true
+      end
 
-    def nosync?
-      @nosync == true
-    end
+      def reset_nosync!
+        remove_instance_variable(:@nosync) if instance_variable_defined?(:@nosync)
+      end
+
+      def nosync(new_value=true)
+        old_value = _nosync
+        self.nosync = new_value
+        yield
+      ensure
+        @nosync = old_value
```

Comment:
Could we make this assign happen via the method like the other lines do with `self.nosync=`? Or if there's a reason this `ensure` needs to directly assign to the instance variable, add a comment to say why? #forfutureus

Reply:
I don't think there's necessarily a reason; I think this is what it _was_ doing before isolating it in the `InstanceMethods`, but I can see about just calling `self.nosync=`. 🤷‍♂